app/assets/javascripts/components/modals/confirmModal.jsx

```diff
@@ -0,0 +1,56 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { actions as appActions } from 'redux/modules/app/index.js';
+import Modal from 'react-modal';
+import SignalIcon from 'components/links/signal_icon.jsx';
+
+
+class ConfirmModal extends Component {
+  constructor(props) {
+    super(props);
+    this.closeModal = this.closeModal.bind(this);
+    this.onConfirm = this.onConfirm.bind(this);
+  }
+
+  componentWillMount() {
+    Modal.setAppElement('body');
+  }
+
+  onConfirm() {
```

Comment:
Ah should these be wrapped in Promises?

I was trying to wrap them, and realized `dispatch` functions already return `Promises`.
Why doesn't `this.props.onConfirm()` return back a promise if it's just a wrapper for a dispatch action?